### DISCUSSION TITLE
Allow a custom value for $maxAttempts in the the Filament\Pages\Auth\Register class' rate limiter

### DISCUSSION THREAD

**Discussion by thierrymaasdam** [filamentphp/filament#13441] [Ideas]
**TLDR**
Allow a custom value for `$maxAttempts` in the the `Filament\Pages\Auth\Register` class' rate limiter

**Current situation**
Within `Filament\Pages\Auth\Register::register()`, the `$maxAttempts` parameter of the rate limiter is now hard-coded to `2`. We have noticed that this causes users to abort their registration with our SaaS product, as they need to wait for a minute after they have made two mistakes in the registration form in a row. It actually happened on a demo call and was slightly awkward.

**Our own solution**
We now have copied the `register()` method on the `Filament\Pages\Auth\Register` class for the sake of setting custom values for the rate limiter.

**Optimal scenario**
Maybe other Filament users would also like to set custom values for the rate limiter. For that reason, I figured it might be useful to drop the feature request here: what if the values for the rate limiter could be set dynamically within the `Filament\Pages\Auth\Register` class?

I look forward to hearing from you. Enjoy your day.

**Comment by ceejayoz** [comment]
I needed this as well.

In my panel provider, I did `->registration(Register::class)` to override the registration form with a custom one, and overrode the `WithRateLimiting` trait. (You can't do this the usual way, because it comes from a trait, so this uses an alias.)

```php
<?php
class Register extends \Filament\Pages\Auth\Register
{
    use WithRateLimiting {
        WithRateLimiting::rateLimit as baseRateLimit;
    }

    protected function rateLimit($maxAttempts, $decaySeconds = 60, $method = null, $component = null): void
    {
        // set the *actual* $maxAttempts rate limit you want here
        $this->baseRateLimit(5, $decaySeconds, $method, $component);
    }
}
```

**Comment by devmount** [comment]
Just for reference: In v4 it is now possible to set the `$maxAttempts` per minute out of the box by using the [`rateLimit()`](https://filamentphp.com/docs/4.x/actions/overview#rate-limiting-actions) method for actions.

```php
use Filament\Actions\Action;

Action::make('register')
    ->rateLimit(5);
```
